geth-utils/gethutil/trace.go

```diff
@@ -158,19 +160,19 @@ func Trace(config TraceConfig) ([]*ExecutionResult, error) {
 			txAccessList[i].Address = accessList.Address
 			txAccessList[i].StorageKeys = accessList.StorageKeys
 		}
-		messages[i] = types.NewMessage(
-			tx.From,
-			tx.To,
-			uint64(tx.Nonce),
-			toBigInt(tx.Value),
-			uint64(tx.GasLimit),
-			toBigInt(tx.GasPrice),
-			toBigInt(tx.GasFeeCap),
-			toBigInt(tx.GasTipCap),
-			tx.CallData,
-			txAccessList,
-			false,
-		)
+		messages[i] = core.Message{
```

Comment:
Instead of this local translation, could consider using upstream `Transaction` type as well as the `TransactionToMessage` helper. It should help with handling EIP-1559 fields better.